packages/thirdweb/src/react/web/ui/ConnectWallet/ConnectButton.tsx

```diff
@@ -410,6 +410,24 @@ function ConnectButtonInner(
     }
   }, [activeAccount]);
 
+  // if the wallet is connected and auth is required, trigger a login attempt automatically
+  useEffect(() => {
+    if (
+      activeAccount &&
```

Comment:
I think we should only do this for in app wallets. Coz on mobile you really need an explicit click otherwise things break quickly.